src/uri/uri.cc

```diff
@@ -240,13 +244,39 @@ auto URI::path() const -> std::optional<std::string> {
     return "/" + this->path_.value();
   }
 
-  size_t path_pos = this->data.find(this->path_.value());
-  if (path_pos != std::string::npos && path_pos > 0 &&
-      this->data[path_pos - 1] == '/') {
-    return "/" + this->path_.value();
+  return path_;
+}
+
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
   }
 
-  return path_;
+  const auto is_relative_path = path.starts_with(".");
+  if (is_relative_path) {
+    throw URIError{"You cannot set a relative path"};
+  }
+
+  auto parsed_path = URI{std::string{path}};
+  this->path_ = parsed_path.path().value();
```

Comment:
```suggestion
  this->path_ = URI{path}.path_;
```

I think this might be enough